java/semantickernel-core/src/main/java/com/microsoft/semantickernel/orchestration/DefaultContextVariables.java

```diff
@@ -148,19 +148,39 @@ public WritableContextVariables build(Map<String, String> map) {
 
     public static class Builder implements ContextVariables.Builder {
 
+        private final Map<String, String> variables;
+
+        public Builder() {
+            variables = new CaseInsensitiveMap<>();
+            this.variables.put(MAIN_KEY, "");
+        }
+
+        @Override
+        public WritableContextVariables buildWritable() {
```

Comment:
I don't think this is needed. One can do `new Builder().build().writableClone()` 
In my opinion, this builder would be better if the API had only this:
```
        ContextVariables build();
        ContextVariables withInput(String input);
        ContextVariables withVariables(Map<String, String> map);
        ContextVariables withVariable(String key, String value);
```